File: language/runtime/android/kotlin/src/main/kotlin/dev/destack/runtime/android/bridge/RuntimeIngress.kt
package dev.destack.runtime.android.bridge

import dev.destack.runtime.android.core.hostStatusNotFound
import dev.destack.runtime.android.core.hostStatusOk

/**
 * Validate one runtime ingress status for one bridge event delivery.
 */
internal fun validateRuntimeIngressStatus(
    status: RuntimeIngressStatus,
    operation: String,
) {
    if (status.code == hostStatusOk || status.code == hostStatusNotFound) {
        return
    }

    error(
        "runtime bridge could not deliver $operation: code ${status.code}, error ${status.errorId}",
    )
}

File: language/runtime/apple/ios/Sources/RuntimeHostIOS/Bridge/Core/RuntimeIngress.swift
import Foundation
import RuntimeHostAppleCore

/// Validate one runtime ingress status for one bridge event delivery.
func validateRuntimeIngressStatus(
  _ status: RuntimeIngressStatus,
  operation: String
) {
  if status.code == hostStatusOk || status.code == hostStatusNotFound {
    return
  }

  preconditionFailure(
    "runtime bridge could not deliver \(operation): code \(status.code), error \(status.errorID)"
  )
}
